### ISSUE TITLE
Nothing notification errors on user creation

### ISSUE THREAD

**Issue by Thoumou** [nextcloud/server#3307]
### Steps to reproduce
1. Go to users panel
2. Create a user with an invalid password like `123456`

### Expected behaviour
A notification with an error message

### Actual behaviour
Nothing notification. An error appears in the JS console

### Server configuration
**Operating system**: Debian testing version

**Web server:**  NGinx

**Database:** MariaDB

**PHP version:** 7.0

**Nextcloud version:** (see Nextcloud admin page) 11.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:**

**Signing status:**
*Nothing*

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.4.1
  - audioplayer: 1.4.1
  - calendar: 1.5.0
  - comments: 1.1.0
  - contacts: 1.5.2
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - news: 10.1.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - ocsms: 1.11.4
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - systemtags: 1.1.3
  - tasks: 0.9.4
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - admin_audit
  - encryption
  - external
  - files_accesscontrol
  - files_automatedtagging
  - files_external
  - files_retention
  - templateeditor
  - user_external
  - user_ldap
  - user_saml
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
list system
{
    "system": {
        "instanceid": "oc13fhcewmn7",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.home.thoumou.fr"
        ],
        "datadirectory": "\/home\/nextcloud\/web\/data",
        "overwrite.cli.url": "http:\/\/nextcloud.home.thoumou.fr",
        "dbtype": "mysql",
        "version": "11.0.1.2",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "maintenance": false
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 50.1

**Operating system:**
Linux Mint 18.1

#### Web server error log

*Nothing*

#### Browser log

![0019 - nextcloud](https://cloud.githubusercontent.com/assets/6215860/22396647/f3520b5a-e55e-11e6-9e73-f4fabc4a22f2.png)

**Comment by nickvergessen** [comment]
Works fine here, please check whether something is in your nextcloud.log file in the data directory

**Comment by Thoumou** [comment]
Sorry but I have no logs about this in `nextcloud.log` and server logs. I have just a 403 for the wrong password request

**Comment by MorrisJobke** [comment]
@Thoumou Is this an admin user or and subadmin user?

**Comment by Thoumou** [comment]
@MorrisJobke The creator is a **SuperAdmin**

**Comment by Thoumou** [comment]
@MorrisJobke Why this reaction ? I have said somthing stupid ?

**Comment by MorrisJobke** [comment]
> @MorrisJobke Why this reaction ? I have said somthing stupid ?

No. I just though that you want to point out, that it's for sure an admin. There is no super admin - we only have admins and subadmin (== group admins).

**Comment by Thoumou** [comment]
Ah ok ! x) I had used this term because the documentation use it too 
https://docs.nextcloud.com/server/11/admin_manual/configuration_user/user_configuration.html#granting-administrator-privileges-to-a-user

**Comment by MorrisJobke** [comment]
> Ah ok ! x) I had used this term because the documentation use it too
> https://docs.nextcloud.com/server/11/admin_manual/configuration_user/user_configuration.html#granting-administrator-privileges-to-a-user

Didn't know that.

**Comment by MorrisJobke** [comment]
@nickvergessen Could this be the "password popup" response and the code somehow doesn't trigger this properly? This works here just fine :/

**Comment by nickvergessen** [comment]
I think it is a server issue, it sends a custom 403 page instead of the generated one.
Because the request to the user management always returns json.

**Comment by MorrisJobke** [comment]
@Thoumou Is there anything in the nextcloud.log?

**Comment by Thoumou** [comment]
@MorrisJobke  No log in nextcloud.log sorry. I use the nginx configuration of this page https://docs.nextcloud.com/server/11/admin_manual/installation/nginx_nextcloud_9x.html.
@nickvergessen The 403 page is not mine, I think it was generates with /core/templates/403.php file.

**Comment by Thoumou** [comment]
Hey ! I found the solution ! On my Nginx configuration I have this line
`error_page 403 /core/templates/403.php;`
And when I try to create a user with a wrong password, the request return a 403.

But why does it return a 403 response for this type of error ? Strange isn't it ?

**Comment by RandieM** [comment]
Same problem here with Debian (7 and 8), nginx 1.6.2, MariaDB (or PostgreSQL), PHP 5.6.30.

Like @Thoumou pointed out, as soon as the line `error_page 403 /core/templates/403.php;` is commented out in the nginx config file, then the notification appears at the very top of the browser window, just like it should.
However, this config line comes from the official Nextcloud documentation, so I am pretty sure that removing it will work around this problem, but break something else.

It would be great if somebody could work on this, as our customers cannot create users in their Nextcloud installations and they have absolutely no clue why, which of course in the end creates a lot of frustration on their side and a lot more work on ours.

Cheers!

**Comment by RandieM** [comment]
Any news on this one?

**Comment by nickvergessen** [comment]
We adjusted the nginx docs to not contain that line anymore: https://github.com/nextcloud/documentation/pull/392
That is the "solution" to this problem, see https://github.com/nextcloud/server/issues/3847

**Comment by RandieM** [comment]
Thanks for your answer @nickvergessen . Unfortunately, the fact that you placed the word solution inside quotes, kind of scares me. Is this some sort of workaround that will "fix" this issue, but cause problems somewhere else?

**Comment by MorrisJobke** [comment]
> Thanks for your answer @nickvergessen . Unfortunately, the fact that you placed the word solution inside quotes, kind of scares me. Is this some sort of workaround that will "fix" this issue, but cause problems somewhere else?

The solution is out of the scope of Nextcloud. That's the reason, why there are the quotes. You need to adjust your Nginx config to fix this issue.

**Comment by nickvergessen** [comment]
Yeah the main problem with the solution is, that people need to find out that they need to adjust their nginx config, since we can't detect that easily.

**Comment by RandieM** [comment]
@MorrisJobke and @nickvergessen, then let me rephrase my question:
What does the line `error_page 403 /core/templates/403.php;` in the nginx config do in relation to Nextcloud? This line has been there for years, because the Nextcloud documentation for nginx called for it.

**Comment by nickvergessen** [comment]
It replaces the site with has 403 + a json response  with a plain html file...
